pkg/kubelet/kubelet.go

```diff
@@ -3076,3 +3118,23 @@ func (kl *Kubelet) warnCgroupV1Usage() {
 	}
 	metrics.CgroupVersion.Set(float64(cgroupVersion))
 }
+
+// staticPodRegistration ensures that all static pods are registered to the apiserver
+// before node became ready.
+func (kl *Kubelet) staticPodsRegistration() error {
+	// kubelet running in standalone mode does not register static pods to the apiserver.
+	if kl.kubeClient == nil {
+		return nil
+	}
+	// Check if initial static pod registration has already been completed.
+	if kl.staticPodsRegistered {
+		return nil
+	}
+	allStaticPods, allMirrorPods := kl.podManager.GetStaticPodsAndMirrorPods()
+	if len(allStaticPods) == len(allMirrorPods) {
+		kl.staticPodsRegistered = true
+		return nil
+	}
+
+	return fmt.Errorf("all static pods are not registered")
```

Comment:
s/all static pods are not registered/not all static pods are registered
for clarity

Reply:
Done.